### PR TITLE
Coverity fixes: the bad

### PR DESCRIPTION
**Pull request by cgwalters**

### CODE REVIEW

File: src/ostree/ot-builtin-pull.c [hunk 1]
```diff
@@ -212,7 +212,7 @@ ostree_builtin_pull (int argc, char **argv, GCancellable *cancellable, GError **
                     override_commit_ids = g_ptr_array_new_with_free_func (g_free);
 
                   /* Backfill */
-                  for (j = 2; j < i; i++)
+                  for (j = 2; j < i; j++)
```
Comment (jlebon) [inline src/ostree/ot-builtin-pull.c]: Ouch, that's a pretty bad one. Let's add a small test case for this as well in `pull-test.sh`? E.g. basically something like this should trigger it, right?

```
${CMD_PREFIX} ostree --repo=parentpullrepo pull origin main@${parent_rev} main@${rev}
```

Comment (cgwalters) [inline src/ostree/ot-builtin-pull.c]: Oh yeah, it was broken.  Needed more work, done in :arrow_down:

**Comment by pwithnall** [comment]
The test case looks good to me.

**Comment by cgwalters** [comment]
@rh-atomic-bot r=pwithnall 5c90602

**Pull request closed without merging**